draft-packages/form/KaizenDraft/Form/Primitives/InputSearch/InputSearch.tsx

```diff
@@ -0,0 +1,104 @@
+import { Icon } from "@kaizen/component-library"
+import { LoadingSpinner } from "@kaizen/draft-loading-spinner"
+import classnames from "classnames"
+import React, { useState } from "react"
+import search from "@kaizen/component-library/icons/search.icon.svg"
+import clear from "@kaizen/component-library/icons/clear.icon.svg"
+
+import styles from "./styles.scss"
+
+export interface InputSearchProps
+  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "className"> {
+  classNameAndIHaveSpokenToDST?: string
+  inputRef?: React.RefObject<HTMLInputElement>
+  reversed?: boolean
+  loading?: boolean
+  primary?: boolean
+  secondary?: boolean
+  startIconAdornment?: React.ReactNode
+  endIconAdornment?: React.ReactNode
+}
+
+const InputSearch: React.FunctionComponent<InputSearchProps> = (
+  props: InputSearchProps
+) => {
+  const {
+    classNameAndIHaveSpokenToDST,
+    placeholder,
+    disabled = false,
+    defaultValue,
+    inputRef,
+    reversed = false,
+    loading,
+    startIconAdornment,
+    endIconAdornment,
+    primary,
+    secondary,
+    ...genericInputProps
+  } = props
+
+  const [value, setValue] = useState("")
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void =>
+    setValue(e.target.value)
+  const handleClear = () => setValue("")
+
+  return (
+    <div
+      className={classnames(
+        styles.wrapper,
+        styles.withStartIconAdornment,
+        styles.withEndIconAdornment,
+        {
+          [styles.withReversed]: reversed,
+          [styles.withDisabled]: disabled,
+          [styles.withSecondary]: secondary,
+          [styles.withLoading]: loading,
+        }
+      )}
+    >
+      <div className={styles.startIconAdornment}>
+        {loading ? (
+          <div className={styles.loadingIcon}>
+            <LoadingSpinner accessibilityLabel="Loading comments" size="sm" />
+          </div>
+        ) : (
+          <Icon icon={search} role="presentation" />
+        )}
+      </div>
+
+      <input
+        type="search"
+        value={value}
+        defaultValue={value ? undefined : defaultValue}
+        ref={inputRef}
+        placeholder={placeholder}
+        disabled={disabled}
+        onChange={handleChange}
+        className={classnames(
+          styles.input,
+          {
+            [styles.default]: !reversed,
+            [styles.reversed]: reversed,
+            [styles.disabled]: disabled,
+            [styles.secondary]: secondary,
+          },
+          classNameAndIHaveSpokenToDST
+        )}
+        {...genericInputProps}
+      />
+      {/* Inputs aren't able to have pseudo elements like ::after on them,
+          so we have to create an element ourselves for the focus ring */}
+      <div className={styles.focusRing} />
```

Comment:
FYI: You can use multiple `box-shadow`'s to create an offset ring as an alternative. The pro is that we aren't having to do this, the con is that the offset color (the 'gap') needs to match the background. Though we have light and dark versions of where these inputs are used anyway so might not be such a con.

Reply:
This has already been implemented in `TextField/Input`, so we'll just keep this for now, but that is good to know!

Reply:
No worries